source/game/field/obj/ObjectKinoko.hh

```diff
@@ -0,0 +1,102 @@
+#pragma once
+
+#include "egg/math/Vector.hh"
+#include "game/field/obj/ObjectCollidable.hh"
+#include "game/field/obj/ObjectKCL.hh"
+
+namespace Field {
+
+enum class KinokoType : u16 {
+    Light = 0,
+    Dark = 1,
+};
+
+/// @brief The class that's responsible for loading the collision data for the mushroom objects
```

Comment:
I think this comment is a bit misleading, it's specifically for the stem of the mushroom
```suggestion
/// @brief The class that represents a mushroom's stem.
```